crates/paymaster-execution/src/diagnostics/extractors/avnu.rs

```diff
@@ -3,63 +3,109 @@
 //! Extracts diagnostic information from failed AVNU swap transactions,
 //! particularly focusing on slippage errors and swap parameters.
 
+use crate::diagnostics::extractors::Metadata;
+use crate::diagnostics::{CallDiagnostic, CallMetadataExtractor, DiagnosticContext, DiagnosticMetric, DiagnosticValue};
+use crate::tokens::TokenClient;
 use async_trait::async_trait;
+use paymaster_starknet::math::felt_to_u128;
+use serde::Serialize;
 use starknet::core::types::{Call, Felt};
-use std::collections::HashMap;
-
-use crate::diagnostics::{CallDiagnostic, CallMetadataExtractor, DiagnosticContext, DiagnosticMetric, DiagnosticValue, ErrorCategory};
-use crate::tokens::TokenService;
+use starknet::macros::selector;
 
 /// AVNU Exchange contract address on Starknet mainnet.
 pub const AVNU_EXCHANGE_ADDRESS: Felt = Felt::from_hex_unchecked("0x04270219d365d6b017231b52e92b3fb5d7c8378b05e9abc97724537a80e93b0f");
+pub const AVNU_EXCHANGE_ADDRESS_SEPOLIA: Felt = Felt::from_hex_unchecked("0x02c56e8b00dbe2a71e57472685378fc8988bba947e9a99b26a00fade2b4fe7c2");
 
-/// Converts a Felt to u128, returning 0 if the value doesn't fit.
-fn felt_to_u128(felt: Felt) -> u128 {
-    felt.to_biguint().try_into().unwrap_or(0)
-}
+/// Slippage exceeded - buy_token_min_amount > buy_token_final_amount
+pub const INSUFFICIENT_TOKENS_RECEIVED: &str = "insufficient tokens received";
+
+/// Slippage exceeded - sell_token_max_amount < sell_token_amount in swap_exact_token_to
+pub const INVALID_TOKEN_MAX_AMOUNT: &str = "invalid token from max amount";
+
+/// User doesn't have enough tokens to sell
+pub const TOKEN_BALANCE_TOO_LOW: &str = "token from balance is too low";
 
-/// Known error messages from the AVNU Exchange contract.
-mod errors {
-    /// Slippage exceeded - buy_token_min_amount > buy_token_final_amount
-    pub const INSUFFICIENT_TOKENS_RECEIVED: &str = "insufficient tokens received";
+/// Token amount is zero
+pub const TOKEN_AMOUNT_ZERO: &str = "token from amount is 0";
 
-    /// Slippage exceeded - sell_token_max_amount < sell_token_amount in swap_exact_token_to
-    pub const INVALID_TOKEN_MAX_AMOUNT: &str = "invalid token from max amount";
+/// Routes array is empty
+pub const ROUTES_EMPTY: &str = "routes is empty";
 
-    /// User doesn't have enough tokens to sell
-    pub const TOKEN_BALANCE_TOO_LOW: &str = "token from balance is too low";
+/// First route sell token doesn't match
+pub const INVALID_TOKEN_FROM: &str = "invalid token from";
 
-    /// Token amount is zero
-    pub const TOKEN_AMOUNT_ZERO: &str = "token from amount is 0";
+/// Last route buy token doesn't match
+pub const INVALID_TOKEN_TO: &str = "invalid token to";
 
-    /// Routes array is empty
-    pub const ROUTES_EMPTY: &str = "routes is empty";
+/// Unknown exchange in routes
+pub const UNKNOWN_EXCHANGE: &str = "unknown exchange";
 
-    /// First route sell token doesn't match
-    pub const INVALID_TOKEN_FROM: &str = "invalid token from";
+pub fn multi_route_swap() -> Felt {
```

Comment:
`selector!` can be used to defined compiled-time values so I'd would propose that we do for example
```
const MULTI_ROUTE_SWAP_SELECTOR: Felt = selector!("multi_route_swap");
```